### PR TITLE
Update top_k_categorical_accuracy to handle 3 dim

### PR DESCRIPTION
**Pull request by simitii**
### Summary
Currently, `top_k_categorical_accuracy` only works with 2 dimentional data : **(batch_size, categories)**
This PR improves `top_k_categorical_accuracy` such that it handles 3 dimentional data such as sequence with shape: **(batch_size, sequence_size, categories)** by first converting the data to shape: **(-1, categories)**

### PR Overview

- [x] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by gabrieldemarmiesse** [comment]
Thanks for the PR. Can you add a unit test? Thanks!

**Comment by simitii** [comment]
@gabrieldemarmiesse unit test added

**Comment by fchollet** [comment]
Thank you for preparing this PR.

We are no longer adding new features to multi-backend Keras, as we are refocusing development efforts on `tf.keras`. If you are still interested in submitting this PR, please direct it to `tf.keras` in the TensorFlow repository instead.
